conpr/08_LE_Tasks/src/le/fj/Sum.java

```diff
@@ -45,8 +47,17 @@ public static void main(String[] args) {
         }
         
         long start = System.currentTimeMillis();
-        int sum = fj.invoke(new Sum(l, 0, l.length));
-        long duration = System.currentTimeMillis() - start;
-        System.out.println("Sum: " + sum + " duration: " + duration + " ms");
+        try {
```

Comment:
Why is this change needed?

Reply:
Because [Future.get()](https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/util/concurrent/Future.html#get()) throws the exceptions in the catch block. The reason I'm now using Future.get() is because originally, the code uses [ForkJoinPool.invoke()](https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/util/concurrent/ForkJoinPool.html#invoke(java.util.concurrent.ForkJoinTask)) -- which returns the result of the task directly -- but [ExecutorService](https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/util/concurrent/ExecutorService.html) doesn't have a method that directly returns the result of a Callable. So I use [ExecutorService.submit(Callable task)](https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/util/concurrent/ExecutorService.html#submit(java.util.concurrent.Callable)) -- which returns a Future representing the result -- and then use get() to get the actual result.